src/unity/python/turicreate/toolkits/object_detector/object_detector.py

```diff
@@ -761,8 +802,8 @@ def _predict_with_options(self, dataset, with_ground_truth,
                                                     dtype=mps_data.dtype)
                         mps_data_padded[:mps_data.shape[0]] = mps_data
                         mps_data = mps_data_padded
-                    self._mps_inference_net.set_input(mps_data)
-                    mps_z = self._mps_inference_net.run_graph()[:n_samples]
+                    self._mps_inference_net.start_batch(mps_data)
```

Comment:
It seems that we always use inference synchronously. Wouldn't it be simpler (or at least more readable) to keep the synchronous option of processing a batch in tcmps - and call it directly?

Reply:
Good catch! Honestly we should be using double buffering for inference, too. But first I'm hoping to replace this API I just changed with a more readable one, modeled after the MXNet API. Starting a batch can return a wrapper around an MPSImage, which triggers a wait for the batch to complete when we read from the image.

For now, I figure there's some value in API consistency across inference and training, too....